packages/block-editor/src/components/inserter/quick-inserter.js

```diff
@@ -163,12 +163,28 @@ function QuickInserter( {
 		[]
 	);
 
+	const clientIdFromInserter = useSelect(
+		( select ) => {
+			const { getBlockIndex, getBlockOrder } = select(
+				'core/block-editor'
+			);
+			// We have to select the previous block because the menu inserter
+			// inserts block after the selected
+			return getBlockOrder( destinationRootClientId )[
+				getBlockIndex( clientId, destinationRootClientId ) - 1
+			];
+		},
+		[ clientId, destinationRootClientId ]
+	);
```

Comment:
Do we really need this, can't we just select `clientId`?

Reply:
When we click the appender button it has as `clientId` the block below the appender, but inserts above the block.
So if we select with `clientId` it will be one block below from where it should be.

I think that was question, right? Or was it about the dependencies array?

Reply:
Oh got it. So is this equivalent to `getPreviousBlockClientId` ?

Reply:
Yes! I didn't know that selector which handles so more things. I'll use it. Thanks!